src/components/Gnb.jsx

```diff
@@ -0,0 +1,59 @@
+import React, { useCallback, useEffect, useState } from "react";
+import { Link } from "react-router-dom";
+import styles from "./Gnb.module.css";
+import logo from "/src/assets/logo.png";
+import { getUserData } from "../api";
+import useAsync from "../hooks/useAsync";
+
+function LoginButton() {
+  return (
+    <Link to="/signin" className={styles.loginBtn}>
+      로그인
+    </Link>
+  );
+}
+
+function UserProfile() {
+  const [user, setUser] = useState(null);
+  const [isApplying, ApplyingError, onApplyAsync] = useAsync(getUserData);
```

Comment:
Apply라는 워딩을 사용한 이유가 궁금합니다.  아래 같은 네이밍들이 더 일반적인것 같아요.
```suggestion
  const [isLoading, error, request] = useAsync(getUserData);
```

Reply:
그냥 코드 작성할 때 직관적으로 생각났던 이름이에요, 플린이 제안한 네이밍이 더 좋은 것 같습니다ㅎㅎ